docs/quickstart.md

```diff
@@ -3,196 +3,186 @@ title: Quickstart
-
-## Query a Hudi dataset
-
-Next, we will register the sample dataset into Hive metastore and try to query using [Hive](#hive), [Spark](#spark) & [Presto](#presto)
+Setup table name, base path and a data generator to generate records for this guide.
 
-### Start Hive Server locally
+```
+import org.apache.hudi.QuickstartUtils._
+import scala.collection.JavaConversions._
+import org.apache.spark.sql.SaveMode._
+import org.apache.hudi.DataSourceWriteOptions._
+import org.apache.hudi.config.HoodieWriteConfig._
+import org.apache.hudi.DataSourceReadOptions._
 
+val tableName = "hudi_cow_table1"
+val basePath = "/tmp/hudi_cow_table1"
+val dataGen = new DataGenerator
 ```
-hdfs namenode # start name node
-hdfs datanode # start data node
 
-bin/hive --service metastore  # start metastore
-bin/hiveserver2 \
-  --hiveconf hive.root.logger=INFO,console \
-  --hiveconf hive.input.format=org.apache.hadoop.hive.ql.io.HiveInputFormat \
-  --hiveconf hive.stats.autogather=false \
-  --hiveconf hive.aux.jars.path=/path/to/packaging/hudi-hive-bundle/target/hudi-hive-bundle-0.4.6-SNAPSHOT.jar
+The [DataGenerator](https://github.com/apache/incubator-hudi/blob/master/hudi-spark/src/main/java/org/apache/hudi/QuickstartUtils.java) 
+can generate sample inserts and updates based on the the sample trip schema [here](#sample-schema)
 
-```
 
-### Run Hive Sync Tool
-Hive Sync Tool will update/create the necessary metadata(schema and partitions) in hive metastore. This allows for schema evolution and incremental addition of new partitions written to.
-It uses an incremental approach by storing the last commit time synced in the TBLPROPERTIES and only syncing the commits from the last sync commit time stored.
-Both [Spark Datasource](writing_data.html#datasource-writer) & [DeltaStreamer](writing_data.html#deltastreamer) have capability to do this, after each write.
+## Write data {#inserts}
+Generate sample records and load them into a DataFrame. Write the DataFrame into the Hudi table as below.
 
 ```
-cd hudi-hive
-./run_sync_tool.sh
-  --user hive
-  --pass hive
-  --database default
-  --jdbc-url "jdbc:hive2://localhost:10010/"
-  --base-path tmp/hoodie/sample-table/
-  --table hoodie_test
-  --partitioned-by field1,field2
+val inserts = convertToStringList(dataGen.generateInserts(10)) 
+val df = spark.read.json(spark.sparkContext.parallelize(inserts, 2));
+df.write.format("org.apache.hudi").options(getQuickstartWriteConfigs).option(PRECOMBINE_FIELD_OPT_KEY, "ts").option(RECORDKEY_FIELD_OPT_KEY, "uuid").option(PARTITIONPATH_FIELD_OPT_KEY, "partitionpath").option(TABLE_NAME, tableName).mode(Overwrite).save(basePath);
+``` 
+You can check the data generated under `/tmp/hudi_cow_table1/<region>/<country>/<city>/`
 
-```
-For some reason, if you want to do this by hand. Please 
-follow [this](https://cwiki.apache.org/confluence/display/HUDI/Registering+sample+dataset+to+Hive+via+beeline).
+**Modelling a record in Hudi:**
```

Comment:
I feel these interfere with the flow.. can we just talk about the key, partition and ts and what they mean briefly and just move over to the update section

Reply:
sure